src/ninja.cc

```diff
@@ -1299,6 +1298,8 @@ NORETURN void real_main(int argc, char** argv) {
     exit((ninja.*options.tool->func)(&options, argc, argv));
   }
 
+  Status* status = new StatusPrinter(config);
```

Comment:
Dealing with raw pointers is prone to memory leaks.

Consider using, when compiled for C++03, std::auto_ptr, and when compiled for C++11 and newer, std::shared_ptr.

Reply:
Better to avoid auto_ptr. I would keep the raw pointer for now.

Reply:
Please elaborate why it's better to avoid using auto_ptr?

auto_ptr is used several places in Ninja's codebase.

Reply:
?

Reply:
?